util/kvencoder/kv_encoder.go

```diff
@@ -77,27 +78,48 @@ type KvEncoder interface {
 	Close() error
 }
 
+var (
+	// refCount is used to ensure that there is only one domain.Domain instance.
+	refCount    int64
```

Comment:
can we remove `refCount`? seems we can use `domGlobal != nil` to replace?

Reply:
No, we can't

Reply:
When refCount decrease to 0, we set `domGlobal = nil`

Reply:
So `refCount == 0` is equal to `domGolbal == nil`?

Reply:
Nope